interfaces/builtin/uinput_test.go

```diff
@@ -0,0 +1,110 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type uinputInterfaceSuite struct {
+	iface        interfaces.Interface
+	coreSlotInfo *snap.SlotInfo
+	coreSlot     *interfaces.ConnectedSlot
+	plugInfo     *snap.PlugInfo
+	plug         *interfaces.ConnectedPlug
+}
+
+var _ = Suite(&uinputInterfaceSuite{
+	iface: builtin.MustInterface("uinput"),
+})
+
+const uinputConsumerYaml = `name: consumer
+version: 0
+apps:
+ app:
+  plugs: [uinput]
+`
+
+const uinputCoreYaml = `name: core
+version: 0
+type: os
+slots:
+  uinput:
+`
+
+func (s *uinputInterfaceSuite) SetUpTest(c *C) {
+	s.plug, s.plugInfo = MockConnectedPlug(c, uinputConsumerYaml, nil, "uinput")
+	s.coreSlot, s.coreSlotInfo = MockConnectedSlot(c, uinputCoreYaml, nil, "uinput")
+}
+
+func (s *uinputInterfaceSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
```

Comment:
minor nitpick, you could make `uinputInterfaceSuite` embed testutil.BaseTest and then do 
```go
s.AddCleanup(func() { dirs.SetRootDir("") })
```

inside `SetUpTest` and not need to define a TearDownTest method